backend/spec/controllers/orgUsersController.spec.js

```diff
@@ -1,28 +1,261 @@
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "",
+        charityNumber: 123456,
+        password: "ABCabc123!",
+      });
+      expect(response.statusCode).toBe(400);
+      expect(response.body).toEqual({
+        error: "Please provide an email address",
+      });
+    });
+
+    test("response code is 400 when password is missing", async () => {
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "puppy@gmail.com",
+        charityNumber: 123456,
+        password: "",
+      });
+      expect(response.statusCode).toBe(400);
+      expect(response.body).toEqual({
+        error: "Please provide a password",
+      });
+    });
+
+    test("response code is 400 when password is weak and does not meet the minimum requirement", async () => {
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "puppy@gmail.com",
+        charityNumber: 123456,
+        password: "abc123",
+      });
+      expect(response.statusCode).toBe(400);
       expect(response.body).toEqual({
-        message: "Thanks! your account has been successfully created",
+        error:
+          "Password must be 8 characters or longer \nPassword must have at least one digit (0-9) \nPassword must have at least one uppercase ('A'-'Z') \nPassword must have at least one special character ('!\"#$%&'()*+,‑./&')",
       });
     });
+
+    test("response code is 201 when password meets all the requirement of a password", async () => {
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "puppy@gmail.com",
+        charityNumber: 123456,
+        password: "ThisIsAStrongPassword1!",
+      });
+      expect(response.statusCode).toBe(201);
+    });
+
+    test("response code is 400 when email address is not valid", async () => {
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "incorrectemail@",
+        charityNumber: 123456,
+        password: "ThisIsAStrongPassword",
+      });
+      expect(response.statusCode).toBe(400);
+      expect(response.body).toEqual({
+        error: "Please provide a valid email address"
+      });
+    });
+
+    test("response code is 201 when email address is valid", async () => {
+      let response = await request(server).post(`/api/orgUsers/signup`).send({
+        organisationName: "Puppies Trust",
+        email: "puppy@gmail.com",
+        charityNumber: 123456,
+        password: "ThisIsAStrongPassword1!",
+      });
+      expect(response.statusCode).toBe(201);
+    });
   });
+
+  describe("Throws error meessages when the client already has an account", () => {
```

Comment:
Small typo in 'messages'

Reply:
ahh messages on purpose as the describe is context - it throw didn't error messages. So I will chance throws to throw